File: leetCode/python/longestCommonPrefix.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        res = ""
        for i in range(len(strs[0])):
            for s in strs:
                if i == len(s) or s[i] != strs[0][i]:
                    return res
            res += strs[0][i]
        return (res)

if __name__== "__main__":
    solution = Solution()
    test1 = ["flower", "flow", "flight"]
    test2 = ["dog", "racecar", "car"]
    test3 = [""]
    print(f"Test 1: {test1} -> {solution.longestCommonPrefix(test1)}")
    print(f"Test 2: {test2} -> {solution.longestCommonPrefix(test2)}")
    print(f"Test 3: {test3} -> {solution.longestCommonPrefix(test3)}")

File: leetCode/C/longestCommonPrefix.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char *longestCommonPrefix(char **strs, int strsSize)
{
	char c;
	int n = 0;

	if (strsSize == 0)
		return "";
	if (strsSize == 1)
		return (strs[0]);

	while ((c = strs[0][n]))
	{
		for (int i = 1; i < strsSize; i++)
		{
			if (strs[i][n] != c)
			{
				strs[0][n] = '\0';
				return (strs[0]);
			}
		}
		n++;
	}
	strs[0][n] = '\0';
	return (strs[0]);
}



int main()
{
	//char *strs[] =  {strdup("flower"), strdup("flow"), strdup("flight")};
	char *strs[] =  {strdup("flow"), strdup("flow"), strdup("flow")};
	char *result = longestCommonPrefix(strs, 3);
	printf("%s", result);

}